api.md

```diff
@@ -280,27 +280,45 @@ with [`add_data_node`](#add_data_node) first before attaching it.
 ---
 ## add_data_node() [](add_data_node)
 
-Add a new data node to the database to be used for creating
-distributed hypertables.
-
-Newly created distributed hypertables will be able to make use of this
+Add a new data node to the database to be used by distributed
+hypertables. This function needs to be executed on the access node by
+a superuser or a user with `USAGE` privileges on the `timescaledb_fdw`
+foreign data wrapper (superuser privileges might still be required on
+the data node). It is assumed that the user role adding a data node
+already has a way to authenticate with the data node being
+added. Alternatively, a password can be provided on the command line
+if the data node accepts password-based authentication and no password
+is provided in a local password file on the access node. Please refer
+to [Setting up Multi-Node TimescaleDB][multinode] for more information
+about node-to-node authentication.
+
+Newly created distributed hypertables will be able to make use of the
 data node for apportioning data across the distributed database. Note
 that existing distributed hypertables will not automatically use the
 newly added data node. For existing distributed hypertables to use
-added data nodes, use [`attach_data_node`](#attach_data_node).
+added data nodes, use [`attach_data_node`](#attach_data_node) after
+having added the data node.
 
 If the data node already exists, the command will abort with either an
 error or a notice depending on the value of `if_not_exists`.
 
-If `bootstrap` is true, the function will attempt to bootstrap the
-data node by:
+Unless `bootstrap` is false, the function will attempt to bootstrap
+the data node by:
 1. Creating the database given in `database` that will serve as the
    new data node.
 2. Loading the TimescaleDB extension in the new database.
 3. Setting metadata to make the data node part of the distributed
    database.
 
-See [boostrapping description][multinode-bootstrap] for details.
+Note that, even if bootstrapping is not requested, the `add_data_node`
+command will try to connect to the remote data node in order to verify
+that it has a valid configuration and that the necessary database
+objects are installed. Distributed user roles are not automatically
+created on the new data node, however, and this might require
+additional setup on the new data node after it is added. The
+[`distributed_exec`](#distributed_exec) procedure can be used to
+create additional database objects on the data node after it is added,
+including distributed roles.
```

Comment:
I think you need to show an example here.

Reply:
There's a link to section that provides examples. I don't think it is appropriate to have examples about one function in a section about another function.

Reply:
Then it might be better to only have a notice that the role needs to exists on the data nodes and then refer to the section about how to make that happen. Right now, you start to talk about something the user needs to do, which suggests that it should finish with an elaboration on how to do that.